HISTORY.md

```diff
@@ -16,6 +16,7 @@
 * Add NewFileChecksumGenCrc32cFactory to the file checksum public API, such that the builtin Crc32c based file checksum generator factory can be used by applications.
 * Add IsDirectory to Env and FS to indicate if a path is a directory.
 * Flush(..., column_family) may return Status::ColumnFamilyDropped() instead of Status::InvalidArgument() if column_family is dropped while processing the flush request.
+* DeleteRange now returns InvalidArgument if the range's begin key is at or after the end key according to the user comparator. Previously the behavior was undefined.
```

Comment:
An empty range (begin == end) was previously undefined? What's wrong with an empty range that we should reject it rather than make it a noop?

Reply:
> An empty range (begin == end) was previously undefined?

I only know about problems for (begin > end). Will update this.

> What's wrong with an empty range that we should reject it rather than make it a noop?

My understanding of what an interval covers comes from wikipedia: `[a, b) = {x | a <= x < b}`. From that perspective, the intervals where `a == b` and `a > b` are both empty. So I thought they can be treated the same.

Reply:
OK, not crazy.

But I suspect passing a > b is more diagnostic of a programming error than a == b. For example, it's common to have loops that might execute zero iterations.

Reply:
If in Math, [3, 3) is valid, allowing it is a good idea. If it doesn't, we can make it invalidate too.

Reply:
I made an update to allow it now. It seems allowed in math. Plus allowing it has the side benefit of reducing unnecessary API change.